### PR TITLE
builtin: add sha/sha1 built-in function

### PR DESCRIPTION
**Pull request by c4pt0r**

### CODE REVIEW

**Comment by shenli** [comment]
@coocood @c4pt0r @ngaut PTAL

File: expression/builtin_encryption.go [hunk 1]
```diff
@@ -533,8 +534,26 @@ type builtinSHA1Sig struct {
...
+		return types.Datum{}, errors.Trace(err)
+	}
+	// SHA/SHA1 function only accept 1 parameter
+	arg := args[0]
+	if arg.IsNull() {
+		return
+	}
+	bin, err := arg.ToBytes()
+	if err != nil {
+		return d, errors.Trace(err)
```
Comment (c4pt0r) [inline expression/builtin_encryption.go]: return types.Datum{}, errors.Trace(err)

Comment (shenli) [inline expression/builtin_encryption.go]: d is equal to types.Datum{}

File: expression/builtin_encryption_test.go [hunk 2]
```diff
@@ -51,6 +51,35 @@ var md5Cases = []md5Test{
 	{nil, nil},
 }
 
+var shaCases = []struct {
+	origin string
+	crypt  string
+}{
+	{"test", "a94a8fe5ccb19ba61c4c0873d391e987982fbbd3"},
+	{"c4pt0r", "034923dcabf099fc4c8917c0ab91ffcd4c2578a6"},
```
Comment (ngaut) [inline expression/builtin_encryption_test.go]: c4pt0r makes me confused.

File: expression/builtin_encryption_test.go [hunk 3]
```diff
@@ -51,6 +51,35 @@ var md5Cases = []md5Test{
...
+	{"pingcap", "73bf9ef43a44f42e2ea2894d62f0917af149a006"},
```
Comment (ngaut) [inline expression/builtin_encryption_test.go]: Can we use "foobar" or something digit string?

**Comment by ngaut** [comment]
LGTM

**Pull request merged**